sample-plugin/pom.xml

```diff
@@ -217,6 +217,11 @@
       <artifactId>antisamy-markup-formatter</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>authentication-tokens</artifactId>
+      <scope>test</scope>
+    </dependency>
```

Comment:
OK, though normally we would prefer to include it implicitly via some other plugin with visible features.

Ideally we would use `dependency:analyze` or similar to verify that this POM includes a minimal set of dependencies.

Reply:
Thanks.  I had not considered that technique.  I think that means that the final conclusion would be to resolve the issue:

* #2096

with pull request

* #2220 

and include the Docker Commons API plugin in the sample plugin.  It depends on authentication-tokens.

Since `docker-commons` is an api plugin without visible features, should we consider adding [Docker Pipeline](https://plugins.jenkins.io/docker-workflow/) to the managed set (110 000 installations) ?

Reply:
Probably yes.

Reply:
> Ideally we would use `dependency:analyze` or similar to verify that this POM includes a minimal set of dependencies

I'm not skilled with maven, but I ran `mvn --pl sample-plugin dependency:analyze` and it reported many declared but unused dependencies.  I tried removing the first one in the list, `jsch`, and that caused `mvn --pl sample-plugin verify` to fail with the report that `jsch` is not listed in the test classpath of the sample plugin.

However, the caffeine-api plugin is not mentioned in the sample-plugin pom file yet is mentioned in the bom-weekly.  Apparently some plugins do not need to be listed in `sample-plugin/pom.xml` but it is not apparent to me which must be listed and which can be skipped.

Clearly I have much more to learn about maven and dependencies.

Reply:
> the caffeine-api plugin is not mentioned in the sample-plugin pom file yet is mentioned in the bom-weekly. Apparently some plugins do not need to be listed in `sample-plugin/pom.xml` but it is not apparent to me which must be listed and which can be skipped.

Those which would be dependencies of other explicitly listed plugins can be omitted, and preferably should (especially in the case of API-only plugins). Whether this is amenable to mechanical enforcement is another question.

> I ran `… dependency:analyze` and it reported many declared but unused dependencies

Unfortunately it is designed for plain old Java programs and there are numerous subtleties specific to Jenkins plugins.